### PR TITLE
Hud10837/import render code

### PR DESCRIPTION
**Pull request by hud10837**
Related to issue: https://devtopia.esri.com/runtime/kotlin/issues/4639

### Description:

Adds rendering code to render the camera feed for Tabletop AR. This will be used by the `TableTopSceneView`.

### Summary of changes:

- Adds a new internal.rendering directory to the module
- Adds several classes related to rendering the camera feed
- Adds a texture and shaders for rendering the camera feed and AR planes
- Adds the ARCore dependency to the ar module

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [ ] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/105/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/render/Mesh.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+     * behave.
+     */
+    public enum PrimitiveMode {
+        POINTS(GLES30.GL_POINTS),
+        LINE_STRIP(GLES30.GL_LINE_STRIP),
+        LINE_LOOP(GLES30.GL_LINE_LOOP),
+        LINES(GLES30.GL_LINES),
+        TRIANGLE_STRIP(GLES30.GL_TRIANGLE_STRIP),
+        TRIANGLE_FAN(GLES30.GL_TRIANGLE_FAN),
+        TRIANGLES(GLES30.GL_TRIANGLES);
```
Comment (hud10837) [inline toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/render/Mesh.java:45]: Most of these are unused, but I think it makes sense to have them as part of a `Mesh` type; we can discuss, maybe this class could be pared down to just rendering the triangle strips

File: toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/render/PlaneRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+    private class SortablePlane(val distance: Float, val plane: Plane)
+
+    public companion object {
+        private val TAG: String = PlaneRenderer::class.java.simpleName
+
+        // Shader names.
+        private const val VERTEX_SHADER_NAME = "shaders/plane.vert"
+        private const val FRAGMENT_SHADER_NAME = "shaders/plane.frag"
+        private const val TEXTURE_NAME = "models/trigrid.png"
```
Comment (gunt0001) [inline toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/render/PlaneRenderer.kt:292]: Is this path correct? Becuase it's under "textures" in the assets folder.

Comment (hud10837) [reply toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/render/PlaneRenderer.kt]: Good catch, I changed the path to textures as we are not using any models

**Comment by hud10837** [comment]
Thanks @gunt0001 I have addressed your comments

**Review by gunt0001 — APPROVED** [review]
@hud10837 - looks good 👍

**Comment by hud10837** [comment]
[Second vtest run](https://runtime-kotlin.esri.com/job/vtest/job/toolkit/111/Test_20Report_20-_20Integration/) looks good, merging now

**Pull request merged**